// pesafy/dist/errors.mjs
//#region src/utils/errors/index.ts
var PesafyError = class PesafyError extends Error {
	code;
	statusCode;
	response;
	requestId;
	cause;
	retryable;
	constructor(options) {
		super(options.message);
		Object.defineProperty(this, "name", { value: "PesafyError" });
		this.code = options.code;
		this.statusCode = options.statusCode;
		this.response = options.response;
		this.requestId = options.requestId;
		this.cause = options.cause;
		this.retryable = options.retryable ?? (options.code === "NETWORK_ERROR" || options.code === "TIMEOUT" || options.code === "RATE_LIMITED" || options.code === "REQUEST_FAILED");
		if (Error.captureStackTrace) Error.captureStackTrace(this, PesafyError);
	}
	/** Returns true if this is a validation error (user bug — do not retry) */
	get isValidation() {
		return this.code === "VALIDATION_ERROR";
	}
	/** Returns true if this is an auth error */
	get isAuth() {
		return this.code === "AUTH_FAILED" || this.code === "INVALID_CREDENTIALS";
	}
	toJSON() {
		return {
			name: this.name,
			code: this.code,
			message: this.message,
			statusCode: this.statusCode,
			requestId: this.requestId,
			retryable: this.retryable
		};
	}
};

//#endregion
export { PesafyError as t };
